### ISSUE TITLE
pigpiod -> 30-35% cpu load on PI 1

### ISSUE THREAD

**Issue by alexschultze** [MuonPi/muondetector#25]
The cpu load is constantly at 30-35% with the default settings for the latest PI OS with only the daemon running.
I assume the configuration is -s 1.

Is the high sample rate actually required?

http://abyz.me.uk/rpi/pigpio/cif.html#gpioCfgClock

**Comment by hangeza** [comment]
Since we are requiring (and through a customized systemctl script as part of the muondetector-daemon package actually starting) pigpiod with the highest possible sampling rate (pigpiod -s 1), the CPU load is as expected and indicated in the link posted by you. The non-default sampling rate which allows a GPIO timing resolution of 1 us is indeed deemed necessary in order to be able to sample the incoming pulses with as low as possible dead time. Currently, the XOR and AND signals which are connected to GPIO pins are evaluated for the determination of the rate, where a loss for signals running in at an average rate of ~4 Hz is quite low due to pile-up, so the standard setting of 5 us sampling time for pgpiod would in principle suffice. Unfortunately, the timing shaper on board of the HAT has a time constant of a liitle bit over 1 us. With the 5us default setting this would leave the majority of the pulses undetected because they would mostly fall between a single clock cycle of the GPIO sampling clock.
Another, not yet implemented, feature which we aim at is the possibility of placing a veto to pulses which are followed at small time intervals by a second pulse. In this case, we would like to know that such events occured to prevent the much slower ADC sampling to acquire the amplitude of the then distorted pulse integral.

Bottom line: the observed CPU load is expected and does not interfere with the operation of the MuonPi. Several detectors have been running constantly for almost a year by now and no side effects or other effects which could be in any connection with pigpiod's CPU consumption were observed.
I therefore tend to close this issue.
Thanks for the feedback.
hgz
